### DISCUSSION TITLE
Polymorphic filter

### DISCUSSION THREAD

**Discussion by ravensnowbird** [AaronLasseigne/active_interaction#549] [Ideas]
I have a generic interaction that takes a object of any class and look at `created_at`  `updated_at` and do some data manipulation return the some value.  Currently my work around is sending a hash 
```
{model_object: @car}
{model_object: @bus}
```

it would be nice to have the filter be able to use ActiveRecord::Object instead of `Car` Or say polymorphic: true.

**Comment by AaronLasseigne** [comment]
You should be able to do this with [object](https://github.com/AaronLasseigne/active_interaction#object):

```ruby
object :obj, class: ActiveRecord::Base
```

Or as an [interface](https://github.com/AaronLasseigne/active_interaction#interface):

```ruby
interface :obj, methods: %i[created_at updated_at]
```

**Comment by antulik** [comment]
For the cases when `interface` approach is too granular you can use extension that adds multiple classes support

https://github.com/antulik/active_interaction-extras#object-multiple-classes
